File: factors.py
#!/usr/bin/python3
from sys import argv

def fac(n: int) -> int:
    for i in range(2, n):
        if n % i == 0:
            return i
    return 1

if len(argv) <= 1:
    print("Usage: factors <file>")
    exit(1)

f = open(argv[1])

line = f.readline()
while line:
    n = int(line)
    q = fac(n)
    p = n // q
    print(f"{n}={p}*{q}")
    line = f.readline()

File: factors.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv)
{
	FILE *f;
	char buf[255];
	int max = 255;
	long long int n;

	if (argc <= 1)
	{
		printf("Usage: factors <file>\n");
		exit(1);
	}

	f = fopen(argv[1], "r");
	while (fgets(buf, max, f) != NULL)
	{
		printf("%lld\n", n);
		n = atoi(buf);
	}

	fclose(f);
	return (0);
}